### PR TITLE
Added api version 2

### PR DESCRIPTION
**Pull request by dapariciom**
Added custom expiration time

### CODE REVIEW

File: src/main/java/com/example/shorturl/controller/api/ShortenUrlControllerV2.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.Objects;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/api/v2/url/user")
```
Comment (BeeAM17) [inline src/main/java/com/example/shorturl/controller/api/ShortenUrlControllerV2.java]: Let's remove user v2 will tell us that this is for logged users

File: src/main/java/com/example/shorturl/controller/api/ShortenUrlControllerV2.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+    @PostMapping
+    public ResponseEntity<UserUrlResponse> shortUrl(@RequestBody UserUrlRequest userUrlRequest) {
+
+        if(Objects.isNull(userUrlRequest) || StringUtils.isEmpty(userUrlRequest.getUrl()))
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Url request is missing or empty");
+
+        Optional<UrlEntity> optionalUrl = shortenUrlService.userShortUrl(userUrlRequest);
+
+        UrlEntity url = optionalUrl.orElseThrow(() -> new ResponseStatusException(HttpStatus.BAD_REQUEST, "Url is missing"));
```
Comment (BeeAM17) [inline src/main/java/com/example/shorturl/controller/api/ShortenUrlControllerV2.java]: Why we need this check? You are trying to short URL and then check if this is URL is empty? :)

File: src/main/java/com/example/shorturl/controller/api/ShortenUrlControllerV2.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+@RequestMapping("/api/v2/url/user")
+public class ShortenUrlControllerV2 {
+
+    private final ShortenUrlService shortenUrlService;
+
+    public ShortenUrlControllerV2(final ShortenUrlService shortenUrlService){
+        this.shortenUrlService = shortenUrlService;
+    }
+
+    @PostMapping
```
Comment (BeeAM17) [inline src/main/java/com/example/shorturl/controller/api/ShortenUrlControllerV2.java:29]: I don't see preauthorize annotation

**Review by BeeAM17 — CHANGES_REQUESTED** [review]
Good work

File: src/main/java/com/example/shorturl/controller/api/ShortenUrlControllerV2.java (hunk 1 above)
Comment (dapariciom) [reply src/main/java/com/example/shorturl/controller/api/ShortenUrlControllerV2.java]: Fixed

File: src/main/java/com/example/shorturl/controller/api/ShortenUrlControllerV2.java (hunk 2 above)
Comment (dapariciom) [reply src/main/java/com/example/shorturl/controller/api/ShortenUrlControllerV2.java]: Fixed on v1 and v2

File: src/main/java/com/example/shorturl/controller/api/ShortenUrlControllerV2.java (hunk 3 above)
Comment (dapariciom) [reply src/main/java/com/example/shorturl/controller/api/ShortenUrlControllerV2.java:29]: Added

File: src/main/java/com/example/shorturl/security/SecurityConfiguration.java [hunk 4]
```diff
@@ -38,7 +43,8 @@ public SecurityFilterChain filterChain(HttpSecurity http) throws Exception{
                 .exceptionHandling().authenticationEntryPoint(jwtAuthEntryPoint).and()
                 .sessionManagement().sessionCreationPolicy(SessionCreationPolicy.STATELESS).and()
                 .authorizeRequests()
-                .antMatchers("/api/**").permitAll()
+                .antMatchers("/api/v1/**").permitAll()
+                .antMatchers("/api/v2/**").hasAnyRole("ADMIN", "USER")
                 .antMatchers("/admin/**").hasRole("ADMIN")
```
Comment (BeeAM17) [inline src/main/java/com/example/shorturl/security/SecurityConfiguration.java:48]: What is the difference between this and line 50?

**Review by BeeAM17 — COMMENTED** [review]
One comment

File: src/main/java/com/example/shorturl/security/SecurityConfiguration.java (hunk 4 above)
Comment (dapariciom) [reply src/main/java/com/example/shorturl/security/SecurityConfiguration.java:48]: I thought that "/admin/**" was not considering "/admin". I checked and it is the same. Let me erase that line

**Review by BeeAM17 — APPROVED** [review]
Let's do some demo tomorrow :)

**Pull request merged**